### PR TITLE
docs: update description of `no-param-reassign`

### PR DESCRIPTION
**Pull request by lumirlumir**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Hello, I've added missing backticks(inline code block) to `no-param-reassign.md`.

In other documentation, `function` and `strict` are wrapped in backticks like the screenshots below.

image

image

### CODE REVIEW

File: docs/src/rules/no-param-reassign.md [hunk 1]
```diff
@@ -6,13 +6,13 @@ further_reading:
 ---
 
 
-Assignment to variables declared as function parameters can be misleading and lead to confusing behavior, as modifying function parameters will also mutate the `arguments` object when not in strict mode (see [When Not To Use It](#when-not-to-use-it) below). Often, assignment to function parameters is unintended and indicative of a mistake or programmer error.
+Assignment to variables declared as `function` parameters can be misleading and lead to confusing behavior, as modifying `function` parameters will also mutate the `arguments` object when not in `strict` mode (see [When Not To Use It](#when-not-to-use-it) below). Often, assignment to `function` parameters is unintended and indicative of a mistake or programmer error.
```
Comment (mdjermanovic) [inline docs/src/rules/no-param-reassign.md:9]: I think "function" shouldn't be in backticks in this context (and the rest of the document) as it doesn't refer to the `function` keyword but any kind of function.

Comment (lumirlumir) [reply docs/src/rules/no-param-reassign.md:9]: Sure, I'll fix it right away.

Then the backticks of `function` in the `description` should be removed too?

image

Comment (mdjermanovic) [reply docs/src/rules/no-param-reassign.md:9]: > Then the backticks of `function` in the `description` section should be removed too?

I think yes.

**Comment by lumirlumir** [comment]
@mdjermanovic I've fixed it and updated `@fileoverview` comment to follow `description` property.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**